portal/models/reporting.py

```diff
@@ -152,10 +152,14 @@ def empro_row_detail(row, ts_reporting):
     # build up data until we find valid cache for patient's history
     status = str(qb_stats.overall_status)
     row = patient_data(patient)
+    row["status"] = status
     if status == "Expired" and research_study_id == EMPRO_RS_ID:
-        row["status"] = "Not Yet Available"
-    else:
-        row["status"] = status
+        # Expired status ambiguous for EMPRO - either not available
+        # due to complex business rules around start or walked off
+        # the end.  Assume if consent + 1year > now, it's the former.
+        consent = datetime.strptime(row["consent"], "%d-%b-%Y %H:%M:%S")
+        if consent + timedelta(days=365) > as_of_date:
```

Comment:
Per https://promowiki.movember.com/pages/viewpage.action?pageId=72941826#SequencingofPROMs-EMPROQUESTIONNAIRE , the actual 12 month expiration date can be significantly later than this.

1. *"If the subject consents to the IRONMAN and EMPRO studies on the same day: the EMPRO questionnaire is made available for four weeks from the date the subject completes their baseline IRONMAN questionnaire"*. So, this cutoff date can be up to 4 weeks too early in this situation.
2. *"If the subject consents to the EMPRO study after consenting to the IRONMAN study... (and) If the subject consents to the EMPRO study more than four weeks after completing an IRONMAN questionnaire: The EMPRO questionnaires are made available for four weeks from the date the subject completes their next IRONMAN questionnaire"* - this may be even longer than four weeks later, as I read it.

Am I reading this correctly? Would we report "Expired" to soon in these situations?

If so, can we look to the timeline for a more accurate date here? If not, we'll need to explain to PCCTC how to interpret this.

Reply:
hmm, yes, you are correct.  i'll need to work that in, thanks for remembering this detail @mcjustin

Reply:
also of note, found in testing scenarios where `row["consent"] == None` need to account for this as well.

Reply:
As I read it, you've addressed this - thanks